lib/index.js

```diff
@@ -279,64 +273,70 @@ Server.prototype.attachServe = function(srv){
 };
 
 /**
- * Handles a request serving `/socket.io.js`
+ * Handles a request serving of client source and map
  *
  * @param {http.Request} req
  * @param {http.Response} res
  * @api private
  */
 
-Server.prototype.serve = function(req, res){
+Server.prototype.serve = function(req, res, url) {
+  var file = url.replace(this._path, '');
+  var isMap = dotMap.test(file);
+  var message = (isMap ? 'map' : 'source');
+
   // Per the standard, ETags must be quoted:
   // https://tools.ietf.org/html/rfc7232#section-2.3
   var expectedEtag = '"' + clientVersion + '"';
 
   var etag = req.headers['if-none-match'];
   if (etag) {
-    if (expectedEtag == etag) {
-      debug('serve client 304');
+    if (expectedEtag === etag) {
+      debug('serve client %s 304', message);
       res.writeHead(304);
       res.end();
       return;
     }
   }
 
-  debug('serve client source');
-  res.setHeader('Content-Type', 'application/javascript');
+  debug('serve client %s', message);
+
+  res.setHeader('Content-Type', 'application/' + (isMap ? 'json' : 'javascript'));
   res.setHeader('ETag', expectedEtag);
-  res.setHeader('X-SourceMap', 'socket.io.js.map');
-  res.writeHead(200);
-  res.end(clientSource);
+
+  if (!isMap) {
+    res.setHeader('X-SourceMap', file.substring(1) + '.map');
+    file = file.replace('.js', '.min.js');
+  }
+
+  this.sendFile(file, req, res);
 };
 
 /**
- * Handles a request serving `/socket.io.js.map`
+ * Writes (and gzippes) a file to the response.
  *
+ * @param {String} file The file to write (relative the client dist directory).
  * @param {http.Request} req
  * @param {http.Response} res
  * @api private
  */
 
-Server.prototype.serveMap = function(req, res){
-  // Per the standard, ETags must be quoted:
-  // https://tools.ietf.org/html/rfc7232#section-2.3
-  var expectedEtag = '"' + clientVersion + '"';
-
-  var etag = req.headers['if-none-match'];
-  if (etag) {
-    if (expectedEtag == etag) {
-      debug('serve client 304');
-      res.writeHead(304);
-      res.end();
-      return;
-    }
+Server.prototype.sendFile = function(file, req, res) {
+  if (!clientRoot) {
+    clientRoot = require.resolve('socket.io-client/dist' + file).replace(file, '/');
   }
 
-  debug('serve client sourcemap');
-  res.setHeader('Content-Type', 'application/json');
-  res.setHeader('ETag', expectedEtag);
-  res.writeHead(200);
-  res.end(clientSourceMap);
+  var readStream = createReadStream(path.join(clientRoot, file));
+  if (~req.headers['accept-encoding'].indexOf('gzip')) {
```

Comment:
I'm not sure this if is necessary.

I don't think we support any browser that doesn't support gzip.